src/Controllers/UserController.js

```diff
@@ -81,30 +81,27 @@ export class UserController extends AdaptableController {
   }
 
   checkResetTokenValidity(username, token) {
-    return this.config.database
-      .find(
-        '_User',
-        {
-          username: username,
-          _perishable_token: token,
-        },
-        { limit: 1 }
-      )
-      .then(results => {
-        if (results.length != 1) {
-          throw 'Failed to reset password: username / email / token is invalid';
-        }
+    let query = {
+      username: username,
+      _perishable_token: token,
+    };
+    if (!token) {
+      query = { $or: [{ email: username }, { username, email: { $exists: false } }] };
+    }
+    return this.config.database.find('_User', query, { limit: 1 }).then(results => {
+      if (results.length != 1) {
+        throw 'Failed to reset password: username / email / token is invalid';
```

Comment:
I believe we should not have a custom message for this case. Someone can you this endpoint to verify if a certain username exists in the database.

Reply:
That is existing functionality [here](https://github.com/parse-community/parse-server/blob/e6ac3b6932186a6a85601c65143305f86f63e948/src/Controllers/UserController.js#L95), all I edited was the `query`, which resulted in lint changes

Reply:
You are now doing an additional query that will lead to the breach.

Reply:
Ok, I understand. I've updated the PR.